plugins/formal-verification/formver.core/src/org/jetbrains/kotlin/formver/names/KotlinName.kt

```diff
@@ -25,7 +26,11 @@ abstract class PrefixedKotlinName(prefix: String, name: Name) : KotlinName {
     override val mangled: String = "${prefix}_${name.asStringStripSpecialMarkers()}"
 }
 
-data class FunctionKotlinName(val name: Name) : PrefixedKotlinName("fun", name)
+abstract class PrefixedKotlinNameWithType(prefix: String, name: Name, type: TypeEmbedding) : KotlinName {
+    override val mangled: String = "${prefix}_${name.asStringStripSpecialMarkers()}${'$'}${type.name.mangled}"
+}
+
+data class FunctionKotlinName(val name: Name, val type: TypeEmbedding) : PrefixedKotlinNameWithType("fun", name, type)
```

Comment:
I wonder if it is worth to only leave the types of arguments for functions without the return type. Maybe the output will be a bit clearer.

Reply:
I'm actually working on a larger overhaul that will let us do stuff like that :)  I sent you review request via Space